mount/mount_id_mapped.go

```diff
@@ -0,0 +1,161 @@
+//go:build linux
+// +build linux
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package mount
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/containerd/containerd/sys"
+	"github.com/sirupsen/logrus"
+)
+
+// TODO: Support multiple mappings in future
+func parseIDMapping(mapping string) ([]syscall.SysProcIDMap, error) {
+	parts := strings.Split(mapping, ":")
+	if len(parts) != 3 {
+		return []syscall.SysProcIDMap{}, fmt.Errorf("user namespace mappings require the format `container-id:host-id:size`")
+	}
+	cID, err := strconv.ParseUint(parts[0], 0, 32)
```

Comment:
I guess it is used for the sake of `ctrID >= 0 && hostID >= 0 && size >= 0`, but true, we can explicitly check that, not sure which way is better though. Later it is used as `int` because fields of that struct are `int`s.

Reply:
@IlyaHanov it will be great if you answer which comments you have addressed. It is very confusing to have to go one by one seeing the diff in another tab, to see if they were addressed.

It seems you didn't do any change for this. Isn't `strconv.Atoi()` give you an int and that is all that we need? You can call validate if you want afterwards too.

Having uint64 is problematic, as that can be a very big number, not sure what happens when you cast it to int (if the number can't be represented).

Reply:
>Having uint64 is problematic, as that can be a very big number, not sure what happens when you cast it to int (if the number can't be represented).

That's true, we can just validate them at the end of the function

Reply:
Done